lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -97,11 +97,19 @@ def add_files_to_build_phases
               }.each do |arc, files|
                 files = files - headers - other_source_files
                 flags = compiler_flags_for_consumer(consumer, arc)
-                regular_file_refs = files.map { |sf| project.reference_for_path(sf) }
+                regular_file_refs = files.map do |sf|
+                  project.reference_for_path(sf).tap do |ref|
+                    raise "unable to find source ref for #{sf} for target #{target.name}" unless ref
+                  end
+                end
                 native_target.add_file_references(regular_file_refs, flags)
               end
 
-              header_file_refs = headers.map { |sf| project.reference_for_path(sf) }
+              header_file_refs = headers.map do |sf|
+                project.reference_for_path(sf).tap do |ref|
```

Comment:
This block looks about the same as above; you could probably wrap this into a private helper method that just takes the name of the kind of file for which you want to print the error ("header" or "source") as an arg.